### PR TITLE
README: Add some documentation about the sandbox

### PR DESCRIPTION
**Pull request by alexlarsson**
This is just some highlevel explainations on how you can sandbox apps.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,7 +32,8 @@ allow privilege escalation.  It may increase the ability of a logged
 in user to perform denial of service attacks, however.
 
 In particular, bubblewrap uses `PR_SET_NO_NEW_PRIVS` to turn off
-setuid binaries.
+setuid binaries, which is the traditional way to get out of things
+like chroots.
```
Comment (cgwalters) [inline README.md]: Looks like <https://en.wikipedia.org/wiki/Chroot#Limitations> mentions this, let's link to it?  There's probably better pages but hey it's wikipedia.  We can edit that section to mention bubblewrap and user namespaces at some point =)

File: README.md [hunk 2]
```diff
@@ -91,6 +92,32 @@ bwrap --ro-bind /usr /usr \
    /bin/sh
 ```
 
+Sandboxing
+----------
+
+The goal of bubblewrap is to run an application in a sandbox, where it has
+access to less resources.
```
Comment (cgwalters) [inline README.md]: s/access to less resources/restricted access to parts of the operating system or user data such as the home directory/ ?

File: README.md [hunk 3]
```diff
@@ -91,6 +92,32 @@ bwrap --ro-bind /usr /usr \
...
+
+bubblewrap always creates a new filesystem namespace, and the user can specify
+exactly what parts of the filesystem should be visible in the sandbox.
+Any such directories you specify mounted nodev by default, and can be made readonly.
```
Comment (cgwalters) [inline README.md]: It's markdown, so `nodev` ?

File: README.md [hunk 4]
```diff
@@ -91,6 +92,32 @@ bwrap --ro-bind /usr /usr \
...
+
+Additionally you can use these kernel features:
+
+User namespaces: This hides all but the current uid and gid from the
```
Comment (cgwalters) [inline README.md]: In this table I'd mention the kernel flags (`CLONE_NEWUSER`).  Also link to e.g. http://linux.die.net/man/2/clone

File: README.md [hunk 5]
```diff
@@ -91,6 +92,32 @@ bwrap --ro-bind /usr /usr \
...
+sandbox. You can also change what the value of uid/gid should be in the sandbox.
+
+IPC namespaces: The sandbox will get its own copy of all the
+different forms of IPCs, like SysV shared memory and semaphores.
+
+PID namespaces: The sandbox will not see any processes outside the sandbox. Additionally, bubblewrap will run a trivial pid1 inside your container to handle the requirements of reaping children in the sandbox.
```
Comment (cgwalters) [inline README.md]: ...This avoids what is known now as the [Docker pid 1 problem](https://blog.phusion.nl/2015/01/20/docker-and-the-pid-1-zombie-reaping-problem/).

File: README.md [hunk 6]
```diff
@@ -91,6 +92,32 @@ bwrap --ro-bind /usr /usr \
...
+
+Network namespaces: The sandbox will not see the network. Instead it will have its own network namespace with only a loopback device.
+
+UTS namespace: The sandbox will have its own hostname.
+
+Seccomp filters: You can pass in seccomp filters that limit which syscalls can be done in the sandbox.
```
Comment (cgwalters) [inline README.md]: For more information, see [Seccomp](https://en.wikipedia.org/wiki/Seccomp).

File: README.md [hunk 7]
```diff
@@ -91,6 +92,35 @@ bwrap --ro-bind /usr /usr \
...
 ```
 
+Sandboxing
+----------
+
+The goal of bubblewrap is to run an application in a sandbox, where it
+has restricted access to parts of the operating system or user data
+such as the home directory.
+
+bubblewrap always creates a new filesystem namespace, and the user can specify
```
Comment (mrunalp) [inline README.md]: Should we use the term "mount namespace" instead of "filesystem namespace"?

Comment (alexlarsson) [inline README.md]: Yeah, that seems to be what the manpage uses, lets do that.

**Comment by alexlarsson** [comment]
I'll merge this now, since its just docs after all and i handled all comments.
@cgwalters r+

**Comment by alexlarsson** [comment]
@cgwalters-bot r+

**Comment by alexlarsson** [comment]
@cgwalters sorry for the wrong r+, but you really need a better name for the bot :)

**Pull request closed without merging**